exporters/otlp/testing-internal/src/main/java/io/opentelemetry/exporter/otlp/testing/internal/ManagedChannelTelemetryExporterBuilder.java

```diff
@@ -182,7 +187,7 @@ public CompletableResultCode export(Collection<T> items) {
 
       @Override
       public CompletableResultCode shutdown() {
-        channel.shutdownNow();
+        Optional.ofNullable(shutdownCallbackRef.get()).ifPresent(Runnable::run);
```

Comment:
this might have been part of my stumbling also though:

```suggestion
        shutdownCallbackRef.get().run();
```